msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/BiGaussian.java

```diff
@@ -0,0 +1,184 @@
+
+
+
+    double interpolationRightSideX = InterpolationX(horizontalSlice, mu, halfHeight, leftBound,
+        rightBound, roundedmz, Direction.RIGHT);
+    // This is sigma right for BiGaussian.
+    sigmaRight = ((interpolationRightSideX - mu)) / Math.sqrt(2 * Math.log(2));
+  }
+
+
+  /**
+   * <p>
+   * InterpolationX is used to calculate X value of Halfwidth-halfheight point of either left or
+   * right half of BiGaussian.
+   * </p>
+   * 
+   * @param mu a {@link java.lang.Integer} object. This is X value for maximum height in terms of
+   *        scan number.
+   * @param halfHeight a {@link java.lang.Double} object. This is m/z value from the raw file.
+   * @param leftBound a {@link java.lang.Integer} object. This is minimum scan number.
+   * @param rightBound a {@link java.lang.Integer} object. This is maximum scan number.
+   * @param roundedmz a {@link java.lang.Integer} object. This is rounded m/z value.
+   * @param direction a {@link Enum} object. This decides for which half we want to calculate X
+   *        value.
+   */
+  private double InterpolationX(MultiKeyMap<Integer, Triplet> horizontalSlice, int mu,
+      double halfHeight, int leftBound, int rightBound, int roundedmz, Direction direction) {
+
+    int i = mu;
+    double Y1 = 0;
+    double Y2 = 0;
+    int step = direction == Direction.RIGHT ? 1 : -1;
+
+    while (leftBound <= i && i <= rightBound) {
+
+      i += step;
+      SliceSparseMatrix.Triplet triplet1 = horizontalSlice.get(i, roundedmz);
+
+      if (triplet1 != null && triplet1.intensity < halfHeight) {
+        Y1 = triplet1.intensity;
+        SliceSparseMatrix.Triplet triplet2 = horizontalSlice.get(i - step, roundedmz);
+        if (triplet2 != null) {
+          Y2 = triplet2.intensity;
+          break;
+        }
+      }
+
+
+
+    }
+
+    /*
+     * I've used the formula of line passing through points (x1,y1) and (x2,y2) in interpolationX.
+     * Those are the points which are exactly above and below of halfHeight(halfMaxIntensity). I've
+     * to find exact point between those two points. I've y-value for that point as halfHeight but I
+     * don't have x-value. X-value is scan number and Y-value is intensity.
+     */
+
+    double X = ((halfHeight - Y1) * ((i - step) - i) / (Y2 - Y1)) + i;
+    return X;
+  }
+
+  /**
+   * <p>
+   * This method is used for getting scan number for given intensity value.
+   * </p>
+   * 
+   * @param height a {@link java.lang.Double} object. This is intensity value from the horizontal
+   *        slice from sparse matrix.
+   */
+  private int getScanNumber(MultiKeyMap<Integer, Triplet> horizontalSlice, double height) {
+    int mu = 0;
+    MapIterator<MultiKey<? extends Integer>, Triplet> iterator = horizontalSlice.mapIterator();
+
+    while (iterator.hasNext()) {
+      iterator.next();
+
+      MultiKey<? extends Integer> mk = (MultiKey<? extends Integer>) iterator.getKey();
```

Comment:
That's odd. Eclipse 4.6.3 doesn't show an warning. What IDE are you using and warning are you getting?